### PR TITLE
Feature/bma/frozen classes

### PR DESCRIPTION
**Pull request by bmarty**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Let danger check for frozen classes

## Motivation and context

First implementation in #6434 was not working correctly, see #6438.

## Tests

- Step 1
- Step 2
- Step ...

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: tools/check/check_code_quality.sh
Comment (ouchadam) [inline tools/check/check_code_quality.sh:23]: are we happy to rely on only danger for this check?

**Comment by bmarty** [comment]
@BillCarsonFr FYI

**Review by ouchadam — APPROVED** [review]
could be nice to have the rule in the local quality check as well but not a blocker

File: tools/check/check_code_quality.sh
Comment (bmarty) [reply tools/check/check_code_quality.sh:23]: since the previous check was not working correctly (see #6438), I think this is fine.
Once merged, I will rebase #6438 to check again that this is working.

**Comment by bmarty** [comment]
> 

There is a very small risk that those files get modified, but could happen during global rework though.

**Comment by bmarty** [comment]
image

So do not wait for the whole CI checks to finish.

**Pull request merged**
